### DISCUSSION TITLE
Problem with discovery in github actions & cibuildwheel

### DISCUSSION THREAD

**Discussion by dsimidzija** [pypa/virtualenv#2349] [General]
As the title says, I've ran into a bit of problem with my github actions, because the cibuildwheel pipeline fails during the virtualenv setup, apparently during the discovery phase. This only happens for osx and windows builds.

This is an example log:

https://github.com/dsimidzija/python-althaia/runs/6741031005?check_suite_focus=true#step%3A5%3A109=

I'm not sure how to fix this, and I couldn't find it being mentioned anywhere in recent virtualenv or cibuildwheel issues. :fearful: If someone can point me in the right direction, I'm willing to help find the cause as much as I can.

**Comment by gaborbernat** [comment]
Seems `/Library/Frameworks/Python.framework/Versions/3.10/bin/python3` prints some text that's not able to be parsed as a JSON. To find out why can you execute with that python this file https://github.com/pypa/virtualenv/blob/main/src/virtualenv/discovery/py_info.py#L1

**Comment by gaborbernat** [comment]
This issue has been fixed in virtualenv 20.15.0+ (released June 2022). PR #2335 added cookie-based markers to isolate JSON output from extraneous text printed by setuptools plugins or other modules during Python discovery, which prevents the "failed to parse JSON" errors in CI environments. Upgrade to virtualenv 20.15.0 or later, and if you're using cibuildwheel, ensure it's using a recent version of virtualenv. Closing as fixed.
